zppy/templates/coupled_global.py

```diff
@@ -1,22 +1,611 @@
+            )
+            FLNT, _ = self.globalAnnualHelper(
+                "FLNT", metric, scale_factor, original_units, final_units
+            )
+            data_array = FSNT - FLNT
+        elif var == "RESTOA":
+            print("NOT READY")
+            FSNTOA, _ = self.globalAnnualHelper(
+                "FSNTOA", metric, scale_factor, original_units, final_units
+            )
+            FLUT, _ = self.globalAnnualHelper(
+                "FLUT", metric, scale_factor, original_units, final_units
+            )
+            data_array = FSNTOA - FLUT
+        elif var == "LHFLX":
+            QFLX, _ = self.globalAnnualHelper(
+                "QFLX", metric, scale_factor, original_units, final_units
+            )
+            PRECC, _ = self.globalAnnualHelper(
+                "PRECC", metric, scale_factor, original_units, final_units
+            )
+            PRECL, _ = self.globalAnnualHelper(
+                "PRECL", metric, scale_factor, original_units, final_units
+            )
+            PRECSC, _ = self.globalAnnualHelper(
+                "PRECSC", metric, scale_factor, original_units, final_units
+            )
+            PRECSL, _ = self.globalAnnualHelper(
+                "PRECSL", metric, scale_factor, original_units, final_units
+            )
+            data_array = (Lv + Lf) * QFLX - Lf * 1.0e3 * (
+                PRECC + PRECL - PRECSC - PRECSL
+            )
+        elif var == "RESSURF":
+            FSNS, _ = self.globalAnnualHelper(
+                "FSNS", metric, scale_factor, original_units, final_units
+            )
+            FLNS, _ = self.globalAnnualHelper(
+                "FLNS", metric, scale_factor, original_units, final_units
+            )
+            SHFLX, _ = self.globalAnnualHelper(
+                "SHFLX", metric, scale_factor, original_units, final_units
+            )
+            LHFLX, _ = self.globalAnnualHelper(
+                "LHFLX", metric, scale_factor, original_units, final_units
+            )
+            data_array = FSNS - FLNS - SHFLX - LHFLX
+        elif var == "PREC":
+            PRECC, _ = self.globalAnnualHelper(
+                "PRECC", metric, scale_factor, original_units, final_units
+            )
+            PRECL, _ = self.globalAnnualHelper(
+                "PRECL", metric, scale_factor, original_units, final_units
+            )
+            data_array = 1.0e3 * (PRECC + PRECL)
+        else:
+            # Non-derived variables
+            if (metric == Metric.AVERAGE) or (metric == Metric.TOTAL):
+                annual_average_dataset_for_var: xarray.core.dataset.Dataset = (
+                    self.f.temporal.group_average(var, "year")
+                )
+                data_array = annual_average_dataset_for_var.data_vars[var]
+            # elif metric == Metric.TOTAL:
+            #     # TODO: Implement this!
+            #     raise NotImplementedError()
```

Comment:
@tomvothecoder @chengzhuzhang I will look into this further, but in case one of you has an immediate answer: 

We currently use `group_average` from XCDAT to calculate the global annual average for variables. The Land team however needs to compute averages of some variables and totals for others. It looks like we don't need to modify the `ts` task then; we just need to modify the calculation here. What I'm not sure of is what calculation to use. Multiply each year's average by a constant (e.g., `landfrac * surface area`)? Use a different xCDAT function?

Reply:
@forsyth2 I believe the formula to compute global sum from global average is to multiply global average by a scalar: the sum of area*landfrac. Maybe you could check if `area` and `landfrac` are available from the `.nc` file generated by `ts` global task.